test/evp_test.c

```diff
@@ -166,12 +156,8 @@ static int test_bin(const char *value, unsigned char **buf, size_t *buflen)
     }
 
     /* Otherwise assume as hex literal and convert it to binary buffer */
-    *buf = OPENSSL_hexstr2buf(value, &len);
-    if (!*buf) {
-        fprintf(stderr, "Value=%s\n", value);
-        ERR_print_errors_fp(stderr);
+    if (!TEST_ptr(*buf = OPENSSL_hexstr2buf(value, &len)))
```

Comment:
The printout of the value is valuable information, and the error print as well.  Most of all, it will otherwise not be printed together with the TEST_ptr error output, but further down, after all tests.